File: netlify/functions/lib/scrapers/nyaa.js
/**
 * Nyaa.si scraper - HTML scraping, primarily for anime.
 */
const client = require('../client');
const { parseSeeders, parseSize, cleanName } = require('../utils');

const NAME = 'Nyaa';
const BASE = 'https://nyaa.si';

async function search(query) {
  const url = `${BASE}/?f=0&c=0_0&q=${encodeURIComponent(query)}&s=seeders&o=desc`;
  const $ = await client.getSoup(url);
  if (!$) return [];

  const results = [];
  const table = $('table.torrent-list');
  if (!table.length) return [];

  table.find('tbody tr').each((_, row) => {
    const cols = $(row).find('td');
    if (cols.length < 7) return;

    const allLinks = $(cols[1]).find('a');
    const nameTag = allLinks.last();
    const name = cleanName(nameTag.text());
    if (!name) return;

    const magnetTag = $(cols[2]).find('a[href^="magnet:"]').first();
    if (!magnetTag.length) return;
    const magnet = magnetTag.attr('href');

    const sizeText = $(cols[3]).text().trim();
    const seeders = parseSeeders($(cols[5]).text());

    results.push({
      name,
      size: sizeText || 'N/A',
      size_gb: parseSize(sizeText),
      seeders,
      magnet,
      provider: NAME,
    });
  });
  return results;
}

module.exports = { name: NAME, base_url: BASE, search };

File: scrapers/nyaa.py
"""Nyaa.si scraper - HTML scraping, primarily for anime content."""

from urllib.parse import quote
from scrapers.base import BaseScraper
from scrapers import client, utils


class Source(BaseScraper):
    name = 'Nyaa'
    base_url = 'https://nyaa.si'

    def search(self, query):
        url = f'{self.base_url}/?f=0&c=0_0&q={quote(query)}&s=seeders&o=desc'
        soup = client.get_soup(url)
        if not soup:
            return []
        results = []
        table = soup.select_one('table.torrent-list')
        if not table:
            return []
        for row in table.select('tbody tr'):
            cols = row.select('td')
            if len(cols) < 7:
                continue
            # Name from second column
            name_tag = cols[1].select('a')[-1] if cols[1].select('a') else None
            name = utils.clean_name(name_tag.get_text()) if name_tag else ''
            if not name:
                continue
            # Magnet from third column
            magnet_tag = cols[2].select_one('a[href^="magnet:"]')
            if not magnet_tag:
                continue
            magnet = magnet_tag['href']
            # Size from fourth column
            size_text = cols[3].get_text(strip=True)
            # Seeders from sixth column
            seeders = utils.parse_seeders(cols[5].get_text())
            results.append({
                'name': name,
                'size': size_text or 'N/A',
                'size_gb': utils.parse_size(size_text),
                'seeders': seeders,
                'magnet': magnet,
                'provider': self.name,
            })
        return results
